### DISCUSSION TITLE
🐛 overrides overrides extends overrides 😂

### DISCUSSION THREAD

**Discussion by unional** [biomejs/biome#3122] [Ideas]
### Environment information

```block
CLI:
  Version:                      1.7.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.0.6"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

```jsonc
// biome.json
{
  "extends": ["./some-config.json"],
  "overrides": [{...}]
}

/// some-config.json
{
  "overrides": [{...}, { "include": ["package.json"], "formatter": { "lineWidth": 20 } }]
}
```

The `overrides` in `biome.json` overrides `overrides` in `some-config.json`

specifically, I observe the `package.json` override does not work until I do:

```jsonc
// biome.json
{
  "extends": ["./some-config.json"],
  "overrides": [{...}, { "include": ["package.json"], "formatter": { "lineWidth": 20 } }]
}
```

### Expected result

The `overrides` in `biome.json` concatenates `overrides` in `some-config.json`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Instead of replacing the inherited overrides, we could extend it by first copying it and the appending the new `overrodes`. This is technically a breaking change. This could be changed in Biome 2.0

**Comment by ematipico** [comment]
I'm not really sure about this... 

Using extends inside overrides was one of the reasons why eslint pushed for a flat config.

Our configuration is already complex, and we should weigh more complexity very carefully and justify it very well.

I am going to close this issue for now because I prefer to not push more complexity on the configuration **as a bug fix**. As mentioned by Vic, it's a possible breaking change.

I will move this issue into a discussion.

**Comment by NexZhu** [comment]
@ematipico In his example there's no `extends` inside `overrides` though. It's about `overrides` in the extended config get replaced completely by the extending config's `overrides`. This disallows shared configs for teams/projects to use `overrides`, IMO it's a big limitation. `overrides` from the extended config should be concatenated by default.
